### ISSUE TITLE
bug: project builds failing after upgrading to 4.20.0

### ISSUE THREAD

**Issue by cyfrost** [ionic-team/stencil#5940]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

4.20.0

### Current Behavior

Hello,

I've recently upgraded a stencil project from v4.19.2 to v4.20.0 after which the builds have started failing.

Here is the build command I've been using always: `stencil build --ci --prod --config=stencil-prod.config.ts`

The error reads `TypeError: Cannot read properties of undefined (reading 'kind') at Object.isClassDeclaration`, I think `node.parent` is undefined while parsing componentMembers. please refer to the screecap below:

<img width="957" alt="image" src="https://github.com/user-attachments/assets/c9c8bfd1-2072-4c7b-b694-65906c31f4d6">

I believe this failure can be attributed to TypeScript upgrade from 5.4.5 to 5.5 (#5898), having checked the compiler's `stencil.js` file by comparing them between the upgrades:

`stencil.js` (v4.19.2)
<img width="1165" alt="image" src="https://github.com/user-attachments/assets/1ca65003-702d-46ac-9943-46e22669f4b2">

`stencil.js` (v4.20.0)
<img width="1340" alt="image" src="https://github.com/user-attachments/assets/74009aa0-beb6-4278-8220-6fe6c75fa12a">

### Expected Behavior

Project build should complete successfully

### System Info

```shell
System: node 20.16.0
    Platform: darwin (23.6.0)
   CPU Model: Apple M1 (8 cpus)
    Compiler: /node_modules/@stencil/core/compiler/stencil.js
       Build: 1722623140
     Stencil: 4.20.0 🚐
  TypeScript: 5.5.3
      Rollup: 2.56.3
      Parse5: 7.1.2
      jQuery: 4.0.0-pre
      Terser: 5.31.1
```

### Steps to Reproduce

This may not be reproducible in a new stencil starter app. However, I'm hoping to understand why this has occurred and if there's a workaround or known solution

### Code Reproduction URL

null

**Comment by cyfrost** [comment]
Just identified the culprit to be a getter called `shadowRoot` in one of the components which is now explicitly disallowed after #5912 

I was able to rename the getter and the builds have since been successful.

Here is how the getter looks now:

```
  private get currentShadowRootRef(): ShadowRoot {
    return this.el.shadowRoot;
  }
```

Closing this, please feel free to reopen if there's any suggestion.

**Comment by christian-bromann** [comment]
A fix for this bug has been published in Stencil [`v4.21.0`](https://github.com/ionic-team/stencil/releases/tag/v4.21.0)!
